server/internal/lsp/server/server.go

```diff
@@ -54,14 +66,25 @@ func NewServer(opts ServerOpts) *Server {
 	}
 
 	handler := protocol.Handler{}
-	glspServer := glspserv.NewServer(&handler, opts.Name, true)
+	glspServer := glspserv.NewServer(&handler, appName, true)
 
 	requestedLanguageVersion := checkRequestedLanguageVersion(opts.C3Version)
 
 	state := l.NewProjectState(logger, option.Some(requestedLanguageVersion.Number), opts.Debug)
 	parser := p.NewParser(logger)
 	search := search.NewSearch(logger, opts.Debug)
-	handlers := handlers.NewHandlers(&state, &parser, search)
+
+	server := &Server{
+		server:  glspServer,
+		options: opts,
+		version: version,
+
+		state:  &state,
+		parser: &parser,
+		search: search,
+
+		diagnosticDebounced: debounce.New(opts.DiagnosticsDelay * time.Millisecond),
```

Comment:
Here I create a debounce function and store it so internal timer is conserved between document changes.